src/lib/FieldType/Type.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformMatrixFieldtype\FieldType;
+
+use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
+use eZ\Publish\Core\FieldType\FieldType;
+use eZ\Publish\Core\FieldType\ValidationError;
+use eZ\Publish\Core\FieldType\Value as FieldTypeValue;
+use eZ\Publish\SPI\FieldType\Value as SPIValue;
+use EzSystems\EzPlatformMatrixFieldtype\FieldType\Value\Row;
+use EzSystems\EzPlatformMatrixFieldtype\FieldType\Value\RowsCollection;
+
+class Type extends FieldType
+{
+    /**
+     * The setting keys which are available on this field type.
+     *
+     * The key is the setting name, and the value is the default value for given
+     * setting, set to null if no particular default should be set.
+     *
+     * @var mixed
+     */
+    protected $settingsSchema = [
+        'minimum_rows' => [
+            'type' => 'integer',
+            'default' => 1,
+        ],
+        'columns' => [
+            'type' => 'hash',
+            'default' => [],
+        ],
+    ];
+
+    /** @var string */
+    private $fieldTypeIdentifier;
+
+    /**
+     * @param string $fieldTypeIdentifier
+     */
+    public function __construct(string $fieldTypeIdentifier)
+    {
+        $this->fieldTypeIdentifier = $fieldTypeIdentifier;
+    }
+
+    /**
+     * Returns information for FieldValue->$sortKey relevant to the field type.
+     *
+     * Return value is mixed. It should be something which is sensible for
+     * sorting.
+     *
+     * It is up to the persistence implementation to handle those values.
+     * Common string and integer values are safe.
+     *
+     * For the legacy storage it is up to the field converters to set this
+     * value in either sort_key_string or sort_key_int.
+     *
+     * In case of multi value, values should be string and separated by "-" or ",".
+     *
+     * @param \eZ\Publish\Core\FieldType\Value $value
+     *
+     * @return mixed
+     */
+    protected function getSortInfo(FieldTypeValue $value)
+    {
+        return '';
+    }
+
+    /**
+     * Validates the fieldSettings of a FieldDefinitionCreateStruct or FieldDefinitionUpdateStruct.
```

Comment:
Adding `{@inheritdoc}`  in all methods from the interface would improve readability of this class greatly.

Reply:
Anonymous classes are part of PHP 7+ and you have to start accepting them... They have feelings too!